translib/transformer/xfmr_file_mgmt.go

```diff
@@ -0,0 +1,113 @@
+//  Copyright 2019 Dell, Inc.                                                 //
+//                                                                            //
+//  Licensed under the Apache License, Version 2.0 (the "License");           //
+//  you may not use this file except in compliance with the License.          //
+//  You may obtain a copy of the License at                                   //
+//                                                                            //
+//  http://www.apache.org/licenses/LICENSE-2.0                                //
+//                                                                            //
+//  Unless required by applicable law or agreed to in writing, software       //
+//  distributed under the License is distributed on an "AS IS" BASIS,         //
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  //
+//  See the License for the specific language governing permissions and       //
+//  limitations under the License.                                            //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+package transformer
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+
+	"github.com/Azure/sonic-mgmt-common/translib/db"
+	"github.com/golang/glog"
+)
+
+var copyMutex = &sync.Mutex{}
+
+func init() {
+	XlateFuncBind("rpc_dir_cb", rpc_dir_cb)
+	XlateFuncBind("rpc_copy_cb", rpc_copy_cb)
+	XlateFuncBind("rpc_delete_cb", rpc_delete_cb)
+	XlateFuncBind("YangToDb_config_reload_key_xfmr", YangToDb_config_reload_key_xfmr)
+}
+
+var rpc_copy_cb RpcCallpoint = func(body []byte, dbs [db.MaxDB]*db.DB) ([]byte, error) {
+	copyMutex.Lock()
+	defer copyMutex.Unlock()
+	return copy_action(body, dbs)
+}
+
+func copy_action(body []byte, dbs [db.MaxDB]*db.DB) ([]byte, error) {
+	var err error
+	var result []byte
+	var options []string
+	var query_result HostResult
+
+	var operand struct {
+		Input struct {
+			Source      string `json:"source"`
+			Destination string `json:"destination"`
+			Copy_option string `json:"copy-config-option"`
+		} `json:"openconfig-file-mgmt-private:input"`
+	}
+
+	var sum struct {
+		Output struct {
+			Status        int32  `json:"status"`
+			Status_detail string `json:"status-detail"`
+		} `json:"openconfig-file-mgmt-private:output"`
+	}
+
+	err = json.Unmarshal(body, &operand)
+
+	if err != nil {
+		/* Unmarshall failed, no input provided.
+		 * set to default */
+		glog.Error("Copy input not provided.")
+		err = errors.New("Input parameters missing.")
+	} else {
+
+		if operand.Input.Source != "running-configuration" || operand.Input.Destination != "startup-configuration" {
+			return nil, errors.New("rpc_copy_cb: Only supports running-configuration -> startup-configuration")
+		}
+
+		glog.Infof("Invoke cfg_mgmt.save %v", options)
+		query_result = HostQuery("cfg_mgmt.save", options)
```

Comment:
I don't think there is any host service module named "cfg_mgmt" in the community branch. Are these changes tested?

Reply:
There isn't cfg_mgmt in the community branch, but that is beyond the scope of this feature.

Reply:
Is this feature even working without this?

Reply:
@rlucus Can you please create an issue to have the host query working for cfg_mgt and assign it to the relevant team and have the issue as part of this description. Also can say that the feature is not complete with this issue.

Reply:
@rlucus I see a host module function to run config save -- https://github.com/sonic-net/sonic-host-services/blob/6fce4781da508709f11be595e3c7048ffd3138cb/host_modules/config_engine.py#L32